src/lib/util/dbuff.h

```diff
@@ -347,6 +368,7 @@ static inline ssize_t fr_dbuff_memcpy_in(fr_dbuff_t *dbuff, uint8_t const *in, s
 
 	fr_assert(!dbuff->is_const);
 
+	_fr_dbuff_bit_flush(dbuff);
```

Comment:
Wouldn't it be better to do this in the advance function?

Reply:
It would. I will make sure that fr_dbuff_bits_in() can work with that.

Reply:
you set remaining bits after calling advance so it should be fine

Reply:
The dictionary parsing code ensures that normal data types starts on a byte boundary.  The first `bit[]` field must also start on a byte boundary.  Then 0 or more other `bit[]` fields.  The final `bit[]` field must end on a byte boundary.

So the rest of the code can assume this is true, and blow up if it isn't true.

Reply:
I'm not sure this is the right approach in general.  Manually specifying the length is annoying.

```
fr_dbuff_bits_in(dbuff, BIT(MY_BIT_MASK_A, value), BIT(MY_BIT_MASK_B, value), BIT(MY_BIT_MASK_C, value))
```

You `|` the masks together, and take the high bit as the length of the bits you're packing. The maximum number of bits would be 64.

I think there'll be cases where we extract things like 4bit fields from structs into their own attributes, and need to re-encode them.  So being able to call fr_dbuff_bits_in and have it continue at a bit offset would still be useful.

Reply:
Seems like there's different requirements here.  I'll probably poke the function once it's pulled in, so don't worry about it for now

Reply:
OK. Aside from the double semicolon, I've left it unchanged.

Reply:
Pushed a change to add FR_DBUFF_BITS_IN_RETURN()